### PR TITLE
[core-js] change Dict interface for strictNullChecks mode

### PR DESCRIPTION
**Pull request by mh-alahdadian**
Dict<T>[string] can have type of undefined and in strictNullChecks mode it should be checked

### CODE REVIEW

File: types/core-js/index.d.ts [hunk 1]
```diff
@@ -337,9 +337,9 @@ declare var log: Log;
 // #############################################################################################
 
 interface Dict<T> {
-    [key: string]: T;
-    [key: number]: T;
-    // [key: symbol]: T;
+    [key: string]: T | undefined;
```
Comment (amcasey) [inline types/core-js/index.d.ts]: Why just these?  Why not (e.g.) `get`, below?

Comment (MHA15) [reply types/core-js/index.d.ts]: some more changes added

Comment (amcasey) [reply types/core-js/index.d.ts]: I don't see an update?

Comment (MHA15) [reply types/core-js/index.d.ts]: my bad, sorry

**Review by amcasey — APPROVED** [review]

**Pull request merged**